registry/api/protocol/eureka/instance_handler.go

```diff
@@ -232,7 +200,7 @@ func (routes *Routes) deregisterInstance(w rest.ResponseWriter, r *rest.Request)
 			"error":     "catalog is nil",
 		}).Errorf("Failed to deregister instance %s", uid)
 
-		i18n.Error(r, w, http.StatusBadRequest, i18n.ErrorNamespaceNotFound)
+		i18n.Error(r, w, http.StatusInternalServerError, i18n.ErrorNamespaceNotFound)
```

Comment:
If a namespace is NOT defined, we creates a new namespace. An error here means some unexpected error occurred. Thus, it is an InternalServerError.